presto-clickhouse/src/main/java/io/prestosql/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -0,0 +1,472 @@
+
+            ImmutableList.Builder<String> columnNames = ImmutableList.builder();
+            ImmutableList.Builder<Type> columnTypes = ImmutableList.builder();
+            ImmutableList.Builder<String> columnList = ImmutableList.builder();
+
+            for (ColumnMetadata column : tableMetadata.getColumns()) {
+                String columnName = column.getName();
+                if (uppercase) {
+                    columnName = columnName.toUpperCase(ENGLISH);
+                }
+                columnNames.add(columnName);
+                columnTypes.add(column.getType());
+                String typeStatement = toWriteMapping(session, column.getType()).getDataType();
+                if (column.isNullable()) {
+                    typeStatement = "Nullable(" + typeStatement + ")";
+                }
+                columnList.add(format("%s %s", columnName, typeStatement));
+            }
+
+            ImmutableList.Builder<String> tableOptions = ImmutableList.builder();
+            ClickHouseTableProperties.getEngine(tableProperties)
+                    .ifPresent(value -> tableOptions.add("ENGINE =" + value));
+
+            String sql = format("CREATE TABLE %s.%s (%s) %s", remoteSchema, tableName, join(", ", columnList.build()),
+                    join(", ", tableOptions.build()));
+            execute(connection, sql);
+            return new JdbcOutputTableHandle(catalog, remoteSchema, remoteTable, columnNames.build(),
+                    columnTypes.build(), Optional.empty(), tableName);
+        }
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName)
+    {
+        execute(session, "CREATE DATABASE " + quoted(schemaName));
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle jdbcColumn,
+            String newColumnName)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            DatabaseMetaData metadata = connection.getMetaData();
+            if (metadata.storesUpperCaseIdentifiers()) {
+                newColumnName = newColumnName.toUpperCase(ENGLISH);
+            }
+            String sql = format("ALTER TABLE %s RENAME COLUMN %s TO %s ",
+                    quoted(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName()),
+                    quoted(jdbcColumn.getColumnName()), quoted(newColumnName));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            // clickhouse versions earlier than 8 do not support the above RENAME COLUMN
+            // syntax
+            if (ClickHouseErrorCode.UNKNOWN_EXCEPTION.toString().equals(e.getSQLState())) {
+                throw new PrestoException(NOT_SUPPORTED,
+                        format("Rename column not supported in catalog: '%s'", handle.getCatalogName()), e);
+            }
```

Comment:
Is Clickhouse < 8 still very popular?

Anyway, i think we can just remove these lines. if it doesn't work, it doesn't work.